### DISCUSSION TITLE
👋 Hey, leave your idea!

### DISCUSSION THREAD

**Discussion by recoluan** [vuepress-reco/vuepress-theme-reco#1] [General]
## Next version, we need your participation! (下一个版本，我们需要你的参与！)

In the past, theme `reco` has been supported by many friends, I am very honored, it changed my life, and I also made a lot of friends. But I don't think it is professional and flexible, and sometimes some advanced features require certain professionalism. I know that this is not easy for some Non-FE professional friends.

I hope the theme can be more out of the box, open and flexible, and inherit "simple" concept, give back to all friends. Come and tell me what you want it to be!

---

在过去一段时间，`reco` 主题已经得到了很多朋友的支持，我感到非常荣幸，它改变了我的生活，同时我也因此结交了很多朋友。但是我觉得它并不能称得上专业和灵活，并且有时候开启一些高级功能是需要一定的专业性的，要知道这对于一些非 FE 专业的朋友并不是容易的。

所以在 2.x 中，我希望主题可以更加开箱即用、开放和灵活，同时继承 “简洁” 的理念，回馈所有朋友，来吧，告诉我你希望的样子！

**Comment by allnan** [comment]
首先感谢大佬做的主题，很实用，简洁，配置也简单

> 我就厚脸皮的当伸手党了

请愿加几个美观实用的callout，vuepress自带的几个markdown it container容器比较简陋，高度也太高
![callout-sample-02.png](https://i.loli.net/2021/04/01/oeT9Xl4hS6j3nRY.png)
![callout-sample-03.png](https://i.loli.net/2021/04/01/ILzmvMawoAeNPn9.png)

**Reply by recoluan** [reply]
这个效果是比较舒服的，采纳，但是实现时间可能会偏后，因为是主题内容的扩展，基于插件可以独立完成。

`reco@2.x` 重点考虑的是将主题配置化，便于主题扩展更多风格，且与主题不耦合，前期还是会专注于主题的核心功能开发，然后主题风格和插件的生态会同时去完善的，希望多多关注。

**Reply by Will-Liang** [reply]
打开右侧子侧边栏，向下滚动页面的时候，回莫名其妙的回到顶部，并且右侧子侧边栏不是fixed

**Comment by ZGuangJu** [comment]
希望首页的样式可以自定义，比如图片，布局等方面，整个主题的颜色，最好能也可以自定，辛苦了！

**Reply by recoluan** [reply]
目前 `reco` 主题 1.0 是支持首页自定义的，当然是需要用户自己去开发一个首页组件，你的建议是维持这种，还是本身提供几种可选风格，可以在配置中去切换呢？
- [ ] A. 用户自己去开发一个首页组件，更加自由；
- [ ] B. 主题提供几种风格，在配置中去切换，依赖性较大。

**Reply by eryajf** [reply]
B，虽然使用主题搭建博客的大多是开发者，但是如果想要面向更多同学，觉得还是多一些模块儿化配置，少一些代码方式配置比较好！

**Reply by penghenger** [reply]
b

**Reply by lty121918** [reply]
a

**Comment by ty-peng** [comment]
- 希望最好能有单图、多图的展示方案支持，还有类似文章封面之类的功能
- 摘要是否可以加一个上边界，有时候 `<!-- more -->` 上方的内容可能不是所有的都想展示在列表上，但是对于文章的结构而言，这部分内容又可能是需要写在前面的
- 三级、四级标题在样式上好像不是特别明显
- 文章列表的时间格式希望也能支持自定义

以上只是一点小建议~

**Reply by recoluan** [reply]
建议很棒，采纳，尽量在 beta 阶段支持，前期会把主要精力放在主题功能的迁移，和主题风格插槽功能的打磨。

**Comment by jinfazhu-mygit** [comment]
建议可以自带一些精美的背景图，伸手党是我没错了(dog)

**Reply by recoluan** [reply]
好的，这个尽力。

**Comment by eryajf** [comment]
- 希望能够针对顶栏导航再做一些设计，主要是针对博客中存在系列文档的场景，目前针对单篇博客已经很优秀了，但是针对于系列文档，希望能给予更直观全面的支持！

**Reply by recoluan** [reply]
我可以理解为是一个专题对吧，对专题有属于自己展示形式。

**Reply by eryajf** [reply]
是的，这里我拿个人博客做个例子，希望不要介意，比如我会写很多[Jenkins相关](http://www.eryajf.net/category/%E6%9C%AF%E4%B8%9A%E4%B8%93%E6%94%BB/%E8%87%AA%E5%8A%A8%E5%8C%96%E8%BF%90%E7%BB%B4/jenkins)的文章，成了一个系列，这个时候按传统博客形式，可能只能像我这样，根据标签或者目录做一个粗略的归类，但是这很不直观。希望能做到那种左侧展示文档目录，中间显示内容这样的一个类似gitbook的展示方式！

**Reply by recoluan** [reply]
那我觉得自定义 sidebar 是不是已经可以满足你的这个要求了？

**Reply by KimYangOfCat** [reply]
> 那我觉得自定义 sidebar 是不是已经可以满足你的这个要求了？

确实可以满足，但是主题自带的博客分类和标签功能会包含所有文档根目录下的所有 markdown 文件，不论文件是否有写 
```
---
tags: 
categories: 
---
```
这就造成系列文章在多个分类中重复的问题，具体描述在 [#370](https://github.com/vuepress-reco/vuepress-theme-reco/issues/370)

**Comment by Fjaxzhy** [comment]
希望可以自定义首页页脚，想多放一点点内容的说

以及感谢大佬的主题，很简洁并且实用

**Reply by recoluan** [reply]
其实页脚不放开的一个最主要的理由就是希望能够通过使用者的网站为主题做一点宣传，建议为主题想一个更好的宣传，从而减少宣传文案对页脚的依赖。

**Reply by ATQQ** [reply]
页脚做增量配置？

**Comment by chenweigao** [comment]
建议代码码的背景颜色可以自定义浅色的~

**Reply by recoluan** [reply]
2.x 内置浅色代码块的背景色，会适应主题配色。

**Reply by chenweigao** [reply]
太棒了~期待 加油！

**Comment by fatFire** [comment]
2.x有没有兴趣使用vitepress来做?

**Reply by recoluan** [reply]
空余时间不足，目前持观望态度，并且vuepress@2.x 目前支持 vite bundler.

**Comment by kuizuo** [comment]
如果可以的话，希望能在查看样式上做些手脚，例如代码的展开与收起，文章封面，博客文章主体卡片化形式等等，目前的话主要还是太偏向于文档类，

**Reply by recoluan** [reply]
采纳

**Comment by hahaxiaowai** [comment]
关于文章的样式，希望能拥有更多的主题选择，如掘金的md一样，在配置中设置即可使用，或能开放一个如同插件广场的地方，让使用者上传自己编辑的样式，供他人使用。

**Reply by recoluan** [reply]
vuepress 本身允许自定义 css，所以我理解功能上可以通过自定义 css 来解决，这个功能更偏向于高度定制，所以暂时不考虑这方面功能的建设。

**Comment by zephyr110** [comment]
建议文章正文视图下的侧边栏添加收放功能，这样在平板类设备上（横、竖屏模式）可以提升阅读体验

**Reply by recoluan** [reply]
采纳

**Comment by gerryge** [comment]
博客首页是不是可以加一个个人说明的地方？现在只是有个头像和名字。

**Reply by recoluan** [reply]
采纳

**Reply by duktig666** [reply]
可以再加一些 个人的常用社交网站链接 点击图标可以跳转那种，例如 CSDN、Github这些

**Comment by snowinszu** [comment]
建议文件和路由结构可以改成 /cn /en  这样，不然外国客户一进来看到的是中文的，如果能根据浏览器语言自动路由到对应的首页那就更好了

**Comment by recoluan** [comment]
> 建议文件和路由结构可以改成 /cn /en  这样，不然外国客户一进来看到的是中文的，如果能根据浏览器语言自动路由到对应的首页那就更好了

采纳，等到开发多语言模块时，会尝试一下这个建议的

**Comment by duktig666** [comment]
如果可以的话，可以出一个【常用链接/官网】页面模板吗？最好还可以对链接进行分类，布局可以紧促点那种。

**Reply by recoluan** [reply]
建议很好，后面有时间会考虑的

**Comment by duktig666** [comment]
博客首页，头像下方可以再加一些 个人的常用社交网站链接 点击图标可以跳转那种，例如 CSDN、Github这些。最好的话可以进行添加配置。

**Reply by duktig666** [reply]
发现这个功能已经有了

**Comment by duktig666** [comment]
## 发现了一款基于vuepress **有很多值得参考的** 网站

网站：[https://www.pdai.tech/](https://www.pdai.tech/)

优点如下：
- **左侧侧边栏的滚动条** 很契合，毫无违和感
- **右侧侧边栏的目录** 非常简介和直观
- **右侧的工具栏** 很nice
- **整体的三栏式布局** 中间有一点空隙间隔（不同颜色），感觉很nice
- **右侧底部的站点图** 也很厉害
- **卡片式的链接导航**

## 部分图片如下
![一个有很多优点的vuepress网站](https://user-images.githubusercontent.com/53132746/157252469-82d9d8f5-7eb8-4ecd-832c-84b1aa9ecb51.png)

![站点图](https://user-images.githubusercontent.com/53132746/157252498-cf19bf52-f39f-4a01-b5df-d97ddb4aa3e4.png)

![链接式导航](https://user-images.githubusercontent.com/53132746/157252562-5ce2f3fc-25a5-4512-8b51-9ee9eed3a762.png)

**Reply by recoluan** [reply]
很好的提议，这些功能可以通过自定义插件实现，并不是主题的必要功能，目前精力有限，还是要紧着主题的核心能力去做，后面再去实现这些个性化能力。

**Comment by yossi-lee** [comment]
希望开放时间线timeline的功能，或者类似hexo的归档页面

**Reply by recoluan** [reply]
> 希望开放时间线timeline的功能，或者类似hexo的归档页面

2023.06.28 正在开发
2023.06.29 开发完，下个版本发布
2023.07.02 2.0.0-beta.66

**Comment by Kaiyuan-Guo** [comment]
1. 希望支持多级目录
2. 支持echarts插件（https://echarts.apache.org/），让Markdown 文件支持图表。
